File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-2.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int l=0,r=0, n=s.size(), ans=0, maxf=0;
        map<char,int> mp;
        while(r<n){
            mp[s[r]]++;
            maxf = max(maxf, mp[s[r]]);
            while(r-l+1-maxf>k){
                mp[s[l]]--;
                maxf = max(maxf,mp[s[l]]);
                l++;
            }
            if(r-l+1-maxf<=k) ans = max(r-l+1,ans);
            r++;
        }
      
        return ans;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        let s= new Set();
        let ans=0;
        for(let num of nums) s.add(num);
        for(let i of s){ 
            let count=0;   
            while(s.has(i++)){ 
                count++;
                // console.log(count); 
                ans = Math.max(ans,count);
            }
        } 
        return ans;
    }
}
